### ISSUE TITLE
Incorrect API Version specified for Microsoft.Compute/cloudServices

### ISSUE THREAD

**Issue by jckeme-rs** [Azure/terraform-provider-azapi#244]
Hi, 

I'm attempting to create a resource with azapi_resource and have specified the core of the terraform configuration similar to how it has been specified here: https://learn.microsoft.com/en-us/azure/templates/microsoft.compute/2022-04-04/cloudservices?pivots=deployment-language-terraform

When this resource gets created however, it always gets created with the API version of 2021-03-01. 

![image](https://user-images.githubusercontent.com/89486915/213300469-3c93c41d-c7bb-4518-ada0-196815f82682.png)

There seems to be no way to force this to use more recent API versions or the one specified in the terraform resource configuration. 

I know though that this "Type" attribute does exist in the terraform state as expected:
```
resource "azapi_resource" "classic_extnd" {
    body                      = (sensitive value)
    id                        = "resource-id-was-here"
    ignore_casing             = true
    ignore_missing_property   = true
    location                  = "location-was-here"
    name                      = "resourcename-was-here"
    output                    = jsonencode({})
    parent_id                 = "parent-id-was-here"
    schema_validation_enabled = false
    tags                      = {}
    type                      = "Microsoft.Compute/cloudServices@2022-04-04"
}

```

Here's what my resource configuration looks like exactly:

```
resource "azapi_resource" "classic_extnd" {
  name = join("-", [var.classic_cloud_service_name, var.environment, format("%03d", var.prefix)])

  parent_id = data.azurerm_resource_group.classic_rsg.id

  type = "Microsoft.Compute/cloudServices@2022-04-04"

  location = var.az_region

  ignore_casing             = true
  ignore_missing_property   = true
  schema_validation_enabled = true

  body = jsonencode({
    properties = {
      configuration      = var.service_configuration_in_xml
      packageUrl         = var.sas_uri_to_service_package_cspkg
      upgradeMode        = "Auto"
      allowModelOverride = var.override_cscfg_sku_properties
      roleProfile = {
        roles = local.list_of_roles
      }
      osProfile = {
        secrets = var.os_profile_secrets_list
      }
      networkProfile = {
        swappableCloudService = null
        loadBalancerConfigurations = [
          {
            name = join("-", [var.classic_cloud_service_name, var.environment, format("%03d", var.prefix), "publicLb"])
            properties = {
              frontendIPConfigurations = [
                {
                  name = join("-", [var.classic_cloud_service_name, var.environment, format("%03d", var.prefix), "pip"])
                  properties = {
                    publicIPAddress = {
                      id = azurerm_public_ip.classic_pub_ip.id
                    }
                  }
                }
              ]
            }
          }
        ]
      }
    }
  })

  tags = merge(var.common_tags, var.custom_tags)

  depends_on = [
    azurerm_public_ip.classic_pub_ip
  ]

}
```

**Comment by ms-henglu** [comment]
Hi @jckeme-rs ,

Thank you for taking time to report this issue!

It seems that you couldn't select the latest api-version from the Portal, right? I guess that api-version is not supported by the Portal yet, but the resource was created by the api-version you specified in azapi provider.

And you can use any api-version with azpai provider as long as the api-version is available.

**Comment by jckeme-rs** [comment]
Hi @ms-henglu 

Thank you for your feedback. 

For the cloudServices resource type, the API version 2022-04-04 does appear to be available.

**From the portal:**
![image](https://user-images.githubusercontent.com/89486915/213342843-d176ed90-43df-4845-8f84-676c82a12ed3.png)

We can also verify the versions on here:
https://learn.microsoft.com/en-us/azure/templates/microsoft.compute/2022-04-04/cloudservices?pivots=deployment-language-terraform

Now I'm mainly trying to interact with this resource using PowerShell but this exact cloudServices resource was created with Terraform + AzAPI. 

The exact error which I am experiencing has also been reported here:
https://stackoverflow.com/questions/75035692/unable-to-update-azure-cloud-service-extended-support-via-azure-powershell-task

I did wonder if there might be an issue with the Powershell Az.CloudService Module, but when I looked at the API version for the resource after it was created, it was always set to **2021-03-01** even after I tainted the resource and re-created it.
